pkg/agent/util.go

```diff
@@ -22,11 +25,26 @@ func getTimestamp() v1Time.Time {
 	return newV1Time
 }
 
-// ApplyResouceToConfig - applies the resources to agent configs
+// ApplyResourceToConfig - applies the resources to agent configs
 // Uses reflection to get the IResourceConfigCallback interface on the config struct or
 // struct variable.
 // Makes call to ApplyResources method with dataplane and agent resources from API server
-func ApplyResouceToConfig(cfg interface{}) error {
+func ApplyResourceToConfig(cfg interface{}) error {
+	obj := cfg
+
+	// This defer func is to catch a possible panic that WILL occur if the cfg object that is passed in embedds the IResourceConfig interface
+	// within its struct, but does NOT implement the ApplyResources method. While it might be that this method really isn't necessary, we will
+	// log an error alerting the user in case it wasn't intentional.
+	defer func() {
+		if err := recover(); err != nil {
+			str := fmt.Sprintf("%v", err)
+			if strings.Contains(str, "nil pointer dereference") {
```

Comment:
rather then the  str := fmt.Sprintf can you do err.Error() in the if statement

Reply:
Yeah. I had Errorf before and did some changes. Will fix.

Reply:
Actually, can't do your suggestion. recover() does not return an error object. It returns an Interface, which under the covers in this case is just a string. Trying to do err.Error() won't compile. So, the %v is the way to basically do a "toString" on the interface.

Reply:
Ahh, did not realize it returned an interface, good call then.